addons/mail/models/mail_message.py

```diff
@@ -903,14 +905,15 @@ def _message_format(self, fnames, format_reply=True):
                 'partners': [{'id': partner.id, 'name': partner.name} for partner in reactions.partner_id],
             } for content, reactions in reactions_per_content.items()]
             allowed_tracking_ids = message_sudo.tracking_value_ids.filtered(lambda tracking: not tracking.field_groups or self.env.is_superuser() or self.user_has_groups(tracking.field_groups))
+            displayed_tracking_ids = thread_record_sudo._track_filter_displayed_tracking_values(allowed_tracking_ids) if thread_record_sudo else allowed_tracking_ids
```

Comment:
Which makes me wonder: how would that happen to have a message and tracking values without model / res_id ? As anyway tracking values should be linked to fields hence to models, maybe we should skip those and avoid any issue / leak with wrong messages ?

Reply:
The idea here was that there isn't a *thread* record, but there should still be a record. It's just not useful if it's not a thread.

We could add an explicit check but in theory if you're able to read a message and the tracking values are not restricted